catroid/src/main/java/org/catrobat/catroid/content/bricks/SetPenColorBrick.java

```diff
@@ -92,7 +91,7 @@ private boolean areAllBrickFieldsNumbers() {
 	}
 
 	private boolean isBrickFieldANumber(BrickField brickField) {
-		return getFormulaWithBrickField(brickField).getRoot().getElementType() == FormulaElement.ElementType.NUMBER;
+		return getFormulaWithBrickField(brickField).getRoot().isElementANumber();
```

Comment:
I don't really understand the `.getRoot()` call here. I'd prefer if classes like this brick here did not have to know about the internal structure of the `Formula` meaning that it'd be better to provide a `isNumber()` function in the `Formula` class. 

An adavantage of that version would be that any changes to the `Formula` would then not require anything to change here.